### PR TITLE
[Discussion] Proposed restructuring

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
⚠️  *This is ready to be looked at and discussed, but, if folks agree with the restructuring plan, it should not be merged until we have a plan for those next steps!* ⚠️ 

### Purpose

I thought I was going to quickly address the rewrite of the transfer learning and multi-task learning sections (#110) and then move onto the putting it all together section (#111), but as I was editing `05.prior-knowledge.md` the representation learning section felt out of place. It occurred to me we talk about dimensionality reduction in the first section (`03.heterogeneity.md`). 

**I propose that we make the section that is about "Managing disparities in data generation is required for robust analysis" more general** – let's talk about issues related to features >> samples, characterizing structure in your data, feature selection and feature learning. By first talking about dimension reduction more generally, we can then focus in on how that can reveal structure that's not of particular scientific interest (batch effects) as a specific motivation or use case. I also think talking about feature selection (_the why_) in the first section could be a helpful bit of set up for some of the points in `04.model-complexity.md`. 

To this end, I've done the following in this pull request:

- Rearranged `03.heterogeneity.md` (no renaming, yet) such that 
  - Some of the ideas expressed above are included as TODOs/HTML comments
  - The text around dimensionality reduction that was specifically about heterogeneity is broken out under the tentative section header `Dimensionality reduction and representation learning`
  - The other text about assessing batch effect (anecdote #113; correction) is in it's own section with the placeholder title `Visualization and correction of technical effects`
- Moved representation learning from `05.prior-knowledge.md` to the section of `03.heterogeneity.md` that also covers dimensionality reduction
- Moved the Dincer et al. and Taroni et al. text that was in `05.prior-knowledge.md` to a new file `06.multiple-approaches-required.md` 
- Renumbered subsequent sections. (Side note: Does "putting it all together" belong in conclusions? I guess we'll find out.)

### Directions for reviewers

The diff for this is not going to be _particularly useful_. For the most part, no small scale text changes of particular consequence have been made. (You will find that the flow is absolutely bonkers as a result 😀 )
Instead, I thought we should discuss this proposed restructuring and then track and split up work such that I am not blocking everyone else.

As such, please focus on the big picture 🖼️  and if you agree with these changes to the structure in principal. If you do agree, do you think the new first section could use its own figure?

### Manuscript checklist

Unless otherwise noted above, this PR will be considered ready for review when all four items have been checked.

- [ 🤞 with all the cutting and pasting ] All changes to text follow "one sentence per line" [[Manubot instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#manubot-markdown)]
- [x] All citations follow the [Manubot citation instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#citations)
- [x] All changes or additions to tables follow the [Manubot table instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#tables)
- [x] All figures follow the [Manubot figure instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#figures)
- [ ] All terms that need to be added to glossary have been added to #50 
- [ ] There are no new spelling errors identified by the Manubot spellchecker

### CODE REVIEW

File: content/05.prior-knowledge.md [hunk 1]
```diff
@@ -46,38 +27,18 @@ Instead, we can build on prior knowledge and large volumes of data that do not d
...
-
-Taken together, DeepProfile [@doi:10.1101/278739] and MultiPLIER [@doi:10.1016/j.cels.2019.04.003] suggest transfer learning can be beneficial for studying rare diseases.
-In the natural images field, researchers have demonstrated that the transferability of features depends on relatedness of tasks [@arxiv:1411.1792].
-The limits of transfer learning for and the concept of relatedness in high-dimensional biomedical data assaying rare diseases are open research questions.
-In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
-
-##### Multitask and few-shot learning
+<!-- TODO: Add a motivating example that's not DeepProfile or MultiPLIER and ties into the first section that I propose to be about dimensionality reduction, etc. :D -->
```
Comment (cgreene) [inline content/05.prior-knowledge.md:34]: could biobombe if you want 🤷

File: content/06.multiple-approaches-required.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+The study by Dincer and colleagues highlights another challenge: samples collected as part of multiple studies may not be associated with the deep phenotypic information that would maximize their scientific value.
+
+
+<!-- From transfer learning -->
+
+Feature-representation-transfer is embodied in Dincer et al., where features are learned from unlabeled AML data and then used to encode a low-dimensional representation of AML data with in vitro drug response labels [@doi:10.1101/278739]. 
+The authors then used this low-dimensional representation as input to predict drug response labels–a supervised example.
+
+In an unsupervised case, Taroni et al. trained a Pathway-Level Information ExtractoR (PLIER) [@doi:10.1038/s41592-019-0456-1] on a large generic collection of human transcriptomic data (recount2 [@doi:10.1038/nbt.3838]) and used the latent variables learned by the model to describe transcriptomic data from the unseen rare diseases antineutrophil cytoplasmic antibody (ANCA)-associated vasculitis (AAV) and medulloblastoma in an approach termed MultiPLIER [@doi:10.1016/j.cels.2019.04.003].
```
Comment (cgreene) [inline content/06.multiple-approaches-required.md]:
```suggestion
In an unsupervised case, Taroni et al. trained Pathway-Level Information ExtractoR (PLIER) [@doi:10.1038/s41592-019-0456-1] on a large generic collection of human transcriptomic data (recount2 [@doi:10.1038/nbt.3838]) and used the latent variables learned by the model to describe transcriptomic data from the unseen rare diseases antineutrophil cytoplasmic antibody (ANCA)-associated vasculitis (AAV) and medulloblastoma in an approach termed MultiPLIER [@doi:10.1016/j.cels.2019.04.003]. 
```

**Review by cgreene — COMMENTED** [review]
I am supportive of this restructuring

**Comment by allaway** [comment]
I too support this proposal. :) 
:shipit: 

I think the first section could use a figure, though might run the risk of veering into "review" territory

**Review by jaybee84 — APPROVED** [review]

**Comment by jaclyn-taroni** [comment]
Everything we wanted to track is now tracked, so I will go ahead and merge this!

**Pull request merged**